lib/puppet/application/agent.rb

```diff
@@ -361,6 +363,16 @@ def run_command
     end
   end
 
+  def log_config
+    #skip also config reading and parsing if debug is not enabled
+    return unless Puppet::Util::Log.level == :debug
```

Comment:
This works, but if there were to be a "higher" log verbosity than `:debug` then it wouldn't work as expected. To check if puppet is at `:debug` or greater, you'd want to use `if Puppet::Util::Log.sendlevel?(:debug)`. That said `:debug` is currently the highest level due to it being index 0 in https://github.com/puppetlabs/puppet/blob/b7e288ca57670c3fbb8153b0f5e42942626e7608/lib/puppet/util/log.rb#L17. So I'll leave it up to you if you want to change or not.

Reply:
I will also update Puppet.debug as it has the same check then and should be consistent.

Reply:
Or better not, as there could be performance decrease